modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java

```diff
@@ -200,6 +200,8 @@ public GridTaskProcessor(GridKernalContext ctx) {
         ctx.io().addMessageListener(TOPIC_TASK_CANCEL, new TaskCancelMessageListener());
         ctx.io().addMessageListener(TOPIC_TASK, new JobMessageListener(true));
 
+        ctx.internalSubscriptionProcessor().registerGlobalStateListener(this);
```

Comment:
`GridTaskProcessor` is a `GridComponent`. Its `onKernalStart()` and `onKernalStop()` are called by `IgniteKernal`. We call twice `onKernalStart()`. Looks wierd. Should it be either `GridComponent` or `IgniteChangeGlobalStateSupport` but not both?

Reply:
In scope of refactoring we shouldn't change logic. It was called twice before changes, perhaps there is a reason for this.